docker/lib/dependabot/docker/utils/credentials_finder.rb

```diff
@@ -32,14 +34,18 @@ def credentials_for_registry(registry_hostname)
           build_aws_credentials(registry_details)
         end
 
+        sig { returns(T.nilable(String)) }
         def base_registry
+          @base_registry ||= T.let(nil, T.nilable(Dependabot::Credential))
           @base_registry ||= credentials.find do |cred|
             cred["type"] == "docker_registry" && cred.replaces_base?
           end
```

Comment:
You can do this in a single statement:

```suggestion
          @base_registry ||= T.let(
          credentials.find do |cred|
            cred["type"] == "docker_registry" && cred.replaces_base?
          end,
          T.nilable(Dependabot::Credential)
          )
```

You'll need to fix the indentation though. The suggestion above is syntactically correct, but Rubocop will complain.